src/challengepaths/2015/CommunityService.ts

```diff
@@ -0,0 +1,372 @@
+    );
+  }
+}
+
+const thralls = new Map<Stat, Thrall>([
+  [$stat`muscle`, $thrall`Undead Elbow Macaroni`],
+  [$stat`moxie`, $thrall`Bind Penne Dreadful`],
+]);
+
+const statTestPredictor = (stat: Stat) => {
+  let baseStat: Stat = stat;
+  if (myClass() === $class`Pastamancer`) {
+    const thrall = thralls.get(stat);
+    if (thrall && myThrall() === thrall) baseStat = $stat`mysticality`;
+  }
+
+  return () =>
+    60 - Math.floor((1 / 30) * (myBuffedstat(stat) - myBasestat(baseStat)));
+};
+
+export const HP = new Test(
+  1,
+  "Donate Blood",
+  () => 60 - Math.floor((myMaxhp() - myBuffedstat($stat`muscle`) - 3) / 30),
+  new Requirement(["HP"], {})
+);
+
+export const Muscle = new Test(
+  2,
+  "Feed The Children",
+  statTestPredictor($stat`muscle`),
+  new Requirement(["Muscle"], {})
+);
+
+export const Mysticality = new Test(
+  3,
+  "Build Playground Mazes",
+  statTestPredictor($stat`mysticality`),
+  new Requirement(["Mysticality"], {})
+);
+
+export const Moxie = new Test(
+  4,
+  "Feed Conspirators",
+  statTestPredictor($stat`moxie`),
+  new Requirement(["Moxie"], {})
+);
+
+export const FamiliarWeight = new Test(
+  5,
+  "Breed More Collies",
+  () =>
+    60 - Math.floor((familiarWeight(myFamiliar()) + weightAdjustment()) / 5),
+  new Requirement(["Familiar Weight"], {})
+);
+
+export const WeaponDamage = new Test(
+  6,
+  "Reduce Gazelle Population",
+  () => {
+    const weaponPower = getPower(equippedItem($slot`weapon`));
+    const offhandPower =
+      toSlot(equippedItem($slot`off-hand`)) === $slot`weapon`
+        ? getPower(equippedItem($slot`off-hand`))
+        : 0;
+    const familiarPower =
+      toSlot(equippedItem($slot`familiar`)) === $slot`weapon`
+        ? getPower(equippedItem($slot`familiar`))
+        : 0;
+    const songDamage =
+      SongBoom.song() === "These Fists Were Made for Punchin'" ? myLevel() : 0;
+    return (
+      60 -
+      Math.floor(
+        (getModifier("Weapon Damage") -
+          0.15 * (weaponPower + offhandPower + familiarPower) -
+          songDamage) /
+          25 +
+          0.001
+      ) -
+      Math.floor(getModifier("Weapon Damage Percent") / 25 + 0.001)
+    );
+  },
+  new Requirement(["Weapon Damage", "Weapon Damage Percent"], {})
+);
+
+export const SpellDamage = new Test(
+  7,
+  "Make Sausage",
+  () => {
+    const spicyDamage = have($skill`Intrinsic Spiciness`)
+      ? Math.min(10, myLevel())
+      : 0;
+    const dragonfishDamage =
+      myFamiliar() === $familiar`Magic Dragonfish`
+        ? numericModifier(
+            $familiar`Magic Dragonfish`,
+            "Spell Damage Percent",
+            familiarWeight($familiar`Magic Dragonfish`) + weightAdjustment(),
+            $item`none`
+          )
+        : 0;
+    return (
+      60 -
+      Math.floor((getModifier("Spell Damage") - spicyDamage) / 25 + 0.001) -
+      Math.floor(
+        (getModifier("Spell Damage Percent") - dragonfishDamage) / 25 + 0.001
```

Comment:
Spell damage and spell damage percent should be divided by 50, not 25, shouldn't it?